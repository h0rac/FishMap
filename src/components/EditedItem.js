import React from 'react';
import { Text, View, StyleSheet, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';

export const EditedItem = (props) =>  {
    
    return (
      <View style={styles.container}>
        <TouchableOpacity style={styles.generalContainer} onPress={()=> props.onEdit(props.selectedPosition.title)}>
        <View style={styles.generalText}>
          <Text style={styles.enableText}>Name:</Text>
          <Text style={styles.textData}>{props.selectedPosition.title}</Text>
        </View>
        </TouchableOpacity>
        <TouchableOpacity style={styles.generalContainer}  onPress={()=> props.onEdit(props.selectedPosition.fishType)}>
        <View style={styles.generalText}>
          <Text style={styles.enableText}>Fish type:</Text>
          <Text style={styles.textData}>{props.selectedPosition.fishType}</Text>
        </View>
        </TouchableOpacity>
        </View>
    )
        
  }
  EditedItem.propTypes = {
      selectedPosition: PropTypes.object,
      onEdit: PropTypes.func
  }

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: 'whitesmoke',
      flexDirection: 'column'
    },
    
    generalContainer: {
      flex: 0.1,
      flexDirection: 'row',
      backgroundColor: 'mintcream',
      alignItems: 'center',
      height: 70,
      marginBottom:2
    },
    
    itemContainer: {
      flex: 0.1,
      flexDirection: 'row',
      backgroundColor: 'mintcream',
      alignItems: 'center',
      paddingBottom: 5,
      paddingTop: 5
    },
    
    section: {
      backgroundColor: 'whitesmoke',
      flex: 0.05,
      paddingLeft: 10,
      paddingTop: 30,
      flexDirection: 'row',
      alignItems: 'center'
    },
    
    enableSwitch: {
      flex: 1,
      flexDirection: 'row',
      justifyContent: 'flex-end',
      paddingRight: 10
    },
    
    sectionText: {
      color: 'gray'
    },
    
    generalText: {
      flex: 3,
      flexDirection: 'row',
      justifyContent: 'flex-start',
      paddingLeft: 10
    },
    
    enableText: {
      paddingTop: 5,
      display:'flex',
      flex:1
    },
  
    textData: {
      paddingTop: 5,
      display:'flex',
      flex:4
    },
    
    title: {
      backgroundColor: 'black',
      color: '#fff',
      fontSize: 18,
      fontWeight: 'bold',
      padding: 10,
      textAlign: 'center'
      }
    });
