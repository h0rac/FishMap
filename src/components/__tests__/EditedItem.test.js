import 'react-native';
import React from 'react';
import {EditedItem} from '../EditedItem';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
describe('Testing EditedItem component', () => {

	let tree;
	const onEdit = jest.fn()
	beforeEach(() => {
        const selectedPosition =  {
            fishType: 'perch',
            title: 'Marker'
        }
        tree = shallow(
		<EditedItem selectedPosition={selectedPosition} onEdit={onEdit} />);
	});
	it('renders correctly', () => {
		expect(tree).toMatchSnapshot();
	});
	it('should trigger edit event', () => {
		const component = tree.dive()
		const touchableOpacities = component.find('TouchableOpacity')
		touchableOpacities.forEach(opacity => {
			opacity.simulate('press')
		})
	})
})