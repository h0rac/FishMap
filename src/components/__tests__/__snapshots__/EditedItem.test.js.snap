// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing EditedItem component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditedItem
    onEdit={[MockFunction]}
    selectedPosition={
      Object {
        "fishType": "perch",
        "title": "Marker",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "mintcream",
              "flex": 0.1,
              "flexDirection": "row",
              "height": 70,
              "marginBottom": 2,
            }
          }
        >
          <Component
            style={
              Object {
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingLeft": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "paddingTop": 5,
                }
              }
            >
              Name:
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "display": "flex",
                  "flex": 4,
                  "paddingTop": 5,
                }
              }
            >
              Marker
            </Text>
          </Component>
        </TouchableOpacity>,
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "mintcream",
              "flex": 0.1,
              "flexDirection": "row",
              "height": 70,
              "marginBottom": 2,
            }
          }
        >
          <Component
            style={
              Object {
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingLeft": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "paddingTop": 5,
                }
              }
            >
              Fish type:
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "display": "flex",
                  "flex": 4,
                  "paddingTop": 5,
                }
              }
            >
              perch
            </Text>
          </Component>
        </TouchableOpacity>,
      ],
      "style": Object {
        "backgroundColor": "whitesmoke",
        "flex": 1,
        "flexDirection": "column",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <Component
            style={
              Object {
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingLeft": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "paddingTop": 5,
                }
              }
            >
              Name:
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "display": "flex",
                  "flex": 4,
                  "paddingTop": 5,
                }
              }
            >
              Marker
            </Text>
          </Component>,
          "onPress": [Function],
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "mintcream",
            "flex": 0.1,
            "flexDirection": "row",
            "height": 70,
            "marginBottom": 2,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "display": "flex",
                    "flex": 1,
                    "paddingTop": 5,
                  }
                }
              >
                Name:
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "display": "flex",
                    "flex": 4,
                    "paddingTop": 5,
                  }
                }
              >
                Marker
              </Text>,
            ],
            "style": Object {
              "flex": 3,
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "paddingLeft": 10,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Name:",
                "ellipsizeMode": "tail",
                "style": Object {
                  "display": "flex",
                  "flex": 1,
                  "paddingTop": 5,
                },
              },
              "ref": null,
              "rendered": "Name:",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Marker",
                "ellipsizeMode": "tail",
                "style": Object {
                  "display": "flex",
                  "flex": 4,
                  "paddingTop": 5,
                },
              },
              "ref": null,
              "rendered": "Marker",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <Component
            style={
              Object {
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingLeft": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "display": "flex",
                  "flex": 1,
                  "paddingTop": 5,
                }
              }
            >
              Fish type:
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "display": "flex",
                  "flex": 4,
                  "paddingTop": 5,
                }
              }
            >
              perch
            </Text>
          </Component>,
          "onPress": [Function],
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "mintcream",
            "flex": 0.1,
            "flexDirection": "row",
            "height": 70,
            "marginBottom": 2,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "display": "flex",
                    "flex": 1,
                    "paddingTop": 5,
                  }
                }
              >
                Fish type:
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "display": "flex",
                    "flex": 4,
                    "paddingTop": 5,
                  }
                }
              >
                perch
              </Text>,
            ],
            "style": Object {
              "flex": 3,
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "paddingLeft": 10,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Fish type:",
                "ellipsizeMode": "tail",
                "style": Object {
                  "display": "flex",
                  "flex": 1,
                  "paddingTop": 5,
                },
              },
              "ref": null,
              "rendered": "Fish type:",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "perch",
                "ellipsizeMode": "tail",
                "style": Object {
                  "display": "flex",
                  "flex": 4,
                  "paddingTop": 5,
                },
              },
              "ref": null,
              "rendered": "perch",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "mintcream",
                "flex": 0.1,
                "flexDirection": "row",
                "height": 70,
                "marginBottom": 2,
              }
            }
          >
            <Component
              style={
                Object {
                  "flex": 3,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "paddingLeft": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "display": "flex",
                    "flex": 1,
                    "paddingTop": 5,
                  }
                }
              >
                Name:
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "display": "flex",
                    "flex": 4,
                    "paddingTop": 5,
                  }
                }
              >
                Marker
              </Text>
            </Component>
          </TouchableOpacity>,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "mintcream",
                "flex": 0.1,
                "flexDirection": "row",
                "height": 70,
                "marginBottom": 2,
              }
            }
          >
            <Component
              style={
                Object {
                  "flex": 3,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "paddingLeft": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "display": "flex",
                    "flex": 1,
                    "paddingTop": 5,
                  }
                }
              >
                Fish type:
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "display": "flex",
                    "flex": 4,
                    "paddingTop": 5,
                  }
                }
              >
                perch
              </Text>
            </Component>
          </TouchableOpacity>,
        ],
        "style": Object {
          "backgroundColor": "whitesmoke",
          "flex": 1,
          "flexDirection": "column",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Component
              style={
                Object {
                  "flex": 3,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "paddingLeft": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "display": "flex",
                    "flex": 1,
                    "paddingTop": 5,
                  }
                }
              >
                Name:
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "display": "flex",
                    "flex": 4,
                    "paddingTop": 5,
                  }
                }
              >
                Marker
              </Text>
            </Component>,
            "onPress": [Function],
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "mintcream",
              "flex": 0.1,
              "flexDirection": "row",
              "height": 70,
              "marginBottom": 2,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "display": "flex",
                      "flex": 1,
                      "paddingTop": 5,
                    }
                  }
                >
                  Name:
                </Text>,
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "display": "flex",
                      "flex": 4,
                      "paddingTop": 5,
                    }
                  }
                >
                  Marker
                </Text>,
              ],
              "style": Object {
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingLeft": 10,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Name:",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "display": "flex",
                    "flex": 1,
                    "paddingTop": 5,
                  },
                },
                "ref": null,
                "rendered": "Name:",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Marker",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "display": "flex",
                    "flex": 4,
                    "paddingTop": 5,
                  },
                },
                "ref": null,
                "rendered": "Marker",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Component
              style={
                Object {
                  "flex": 3,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "paddingLeft": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "display": "flex",
                    "flex": 1,
                    "paddingTop": 5,
                  }
                }
              >
                Fish type:
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "display": "flex",
                    "flex": 4,
                    "paddingTop": 5,
                  }
                }
              >
                perch
              </Text>
            </Component>,
            "onPress": [Function],
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "mintcream",
              "flex": 0.1,
              "flexDirection": "row",
              "height": 70,
              "marginBottom": 2,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "display": "flex",
                      "flex": 1,
                      "paddingTop": 5,
                    }
                  }
                >
                  Fish type:
                </Text>,
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "display": "flex",
                      "flex": 4,
                      "paddingTop": 5,
                    }
                  }
                >
                  perch
                </Text>,
              ],
              "style": Object {
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingLeft": 10,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Fish type:",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "display": "flex",
                    "flex": 1,
                    "paddingTop": 5,
                  },
                },
                "ref": null,
                "rendered": "Fish type:",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "perch",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "display": "flex",
                    "flex": 4,
                    "paddingTop": 5,
                  },
                },
                "ref": null,
                "rendered": "perch",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
